pyomo/dae/flatten.py

```diff
@@ -200,8 +200,28 @@ def slice_component_along_sets(component, sets, context_slice=None, normalize=No
                     #
                     # Note that c_slice is not necessarily a slice.
                     # We enter this loop even if no sets need slicing.
-                    temp_slice = c_slice.duplicate()
-                    next(iter(temp_slice))
+                    try:
+                        next(iter(c_slice.duplicate()))
+                    except IndexError:
+                        if normalize_index.flatten:
+                            raise
+                        # There is an edge case where when we are not
```

Comment:
I would recommend adding a test for this edge case, if possible, considering that we explicitly check for it.

Reply:
Poke again on this. Feel free to say "No"

Reply:
I think I am going to go with "No".  I am pretty sure that the case I was describing is already being covered by tests - the comment was mostly a note to our future selves why this was implemented the way it was.